waspls/src/Wasp/Backend/ConcreteParser.hs

```diff
@@ -0,0 +1,98 @@
+module Wasp.Backend.ConcreteParser
+  ( -- * Concrete parser
+
+    -- | This module contains functions for converting lists of "Token"s into
+    -- concrete syntax trees. These trees represent any source file completely,
+    -- including whitespace and comments, even if the source file is invalid.
+    parseCST,
+    parseCSTExpression,
+  )
+where
+
+import Wasp.Backend.ConcreteParser.Internal
+import Wasp.Backend.ConcreteSyntax (SyntaxKind (..), SyntaxNode)
+import Wasp.Backend.ParseError
+import Wasp.Backend.Token (Token)
+import qualified Wasp.Backend.Token as T
+
+-- | Parse a list of tokens into a concrete syntax tree for an entire wasp file.
+--
+-- See "SyntaxNode" for a description of what a concrete syntax tree contains.
+parseCST :: [Token] -> ([ParseError], [SyntaxNode])
+parseCST tokens = parse tokens (root --> eof)
+
+-- | Parse a list of tokens into a concrete syntax tree for a wasp expression.
+-- This is mainly used for testing.
+--
+-- See "SyntaxNode" for a description of what a concrete syntax tree contains.
+parseCSTExpression :: [Token] -> ([ParseError], [SyntaxNode])
+parseCSTExpression tokens = parse tokens (expr --> eof)
```

Comment:
I agree with what you wrote! I do like `-->` on one hand, it is visually attractive :D, but it keeps intuitively somehow pushing me in the wrong direction, and I think `<>` might be a more natural way to go, especially since it gives us mconcat also, and at the end it is nice to have Semigroup and Monoid defined.

The last line, which is parser generator style -> I don't think you can achieve that easily, nor would it look good like this in Haskell, I mean it looks like a bunch of application. Maybe, if you would do the list thing: `[kwImport, extImportName, ...]`, but if we go with Monoid we get that for free with mconcat!